internal/fxlog/fxlog.go

```diff
@@ -21,68 +21,117 @@
 package fxlog
 
 import (
-	"fmt"
-	"log"
-	"os"
-	"strings"
-
-	"go.uber.org/fx/internal/fxreflect"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
-var _exit = func() { os.Exit(1) }
+// Level is the level of logging used by Logger.
+type Level int
+
+const (
+	// InfoLevel is Info logging level used to log messages via zap.
+	InfoLevel Level = iota
+	// ErrorLevel is Info logging level used to log messages via zap.
+	ErrorLevel
+)
 
-// Printer is a formatting printer.
-type Printer interface {
-	Printf(string, ...interface{})
+// Entry is an entry to be later serialized into zap message and fields.
+type Entry struct {
+	Level   Level
+	Message string
+	Fields  []Field
+	Stack   string
 }
 
-// New returns a new Logger backed by the standard library's log package.
-func New() *Logger {
-	return &Logger{log.New(os.Stderr, "", log.LstdFlags)}
+// WithStack mutates an Entry to add a Stack field.
+func (e Entry) WithStack(stack string) Entry {
+	e.Stack = stack
+	return e
 }
 
-// A Logger writes output to standard error.
-type Logger struct {
-	Printer
+func (e Entry) Write(logger Logger) {
+	logger.Log(e)
 }
 
-// Printf logs a formatted Fx line.
-func (l *Logger) Printf(format string, v ...interface{}) {
-	l.Printer.Printf(prepend(format), v...)
+// Err is a helper for error Fields.
+func Err(value interface{}) Field {
```

Comment:
shouldn't this always be an error

```suggestion
func Err(value error) Field {
```